internal/controller/controller.go

```diff
@@ -248,13 +248,17 @@ func backendSecurityPolicyIndexFunc(o client.Object) []string {
 	switch backendSecurityPolicy.Spec.Type {
 	case aigv1a1.BackendSecurityPolicyTypeAPIKey:
 		apiKey := backendSecurityPolicy.Spec.APIKey
-		key = getSecretNameAndNamespace(apiKey.SecretRef, backendSecurityPolicy.Namespace)
+		if apiKey != nil {
+			key = getSecretNameAndNamespace(apiKey.SecretRef, backendSecurityPolicy.Namespace)
+		}
 	case aigv1a1.BackendSecurityPolicyTypeAWSCredentials:
 		awsCreds := backendSecurityPolicy.Spec.AWSCredentials
-		if awsCreds.CredentialsFile != nil {
-			key = getSecretNameAndNamespace(awsCreds.CredentialsFile.SecretRef, backendSecurityPolicy.Namespace)
-		} else if awsCreds.OIDCExchangeToken != nil {
-			key = backendSecurityPolicyKey(backendSecurityPolicy.Namespace, backendSecurityPolicy.Name)
+		if awsCreds != nil {
```

Comment:
can you fix this at the k8s layer level rather than adding unnecessary if block? when the type is AWSBedrock then this should not exist in the first place.

Reply:
i mean: add a new CEL validation and corresponding test cases to tests/crdcel and revert this if block